src/modules/robot/arm_solutions/RotatableDeltaSolution.cpp

```diff
@@ -0,0 +1,251 @@
+	float alpha_theta = 0.0F;
+	float beta_theta  = 0.0F;
+	float gamma_theta = 0.0F;
+
+	//Code from Trossen Robotics tutorial, note we put the X axis at the back and not the front of the robot.
+
+	float x0 = cartesian_mm[X_AXIS];
+	float y0 = cartesian_mm[Y_AXIS];
+	float z_with_offset = cartesian_mm[Z_AXIS] + z_calc_offset; //The delta calculation below places zero at the top.  Subtract the Z offset to make zero at the bottom.
+
+	int status =              delta_calcAngleYZ(x0,                    y0,                  z_with_offset, alpha_theta);
+	if (status == 0) status = delta_calcAngleYZ(x0*cos120 + y0*sin120, y0*cos120-x0*sin120, z_with_offset, beta_theta);  // rotate co-ordinates to +120 deg
+	if (status == 0) status = delta_calcAngleYZ(x0*cos120 - y0*sin120, y0*cos120+x0*sin120, z_with_offset, gamma_theta);  // rotate co-ordinates to -120 deg
+
+	if (status == -1)  //something went wrong
+	{
+//DEBUG CODE, uncomment the following to help determine what may be happening if you are trying to adapt this to your own different roational delta.
+//	    THEKERNEL->streams->printf("ERROR: Delta calculation fail!  Unable to move to:\n");
+//	    THEKERNEL->streams->printf("    x= %f\n",cartesian_mm[X_AXIS]);
+//	    THEKERNEL->streams->printf("    y= %f\n",cartesian_mm[Y_AXIS]);
+//	    THEKERNEL->streams->printf("    z= %f\n",cartesian_mm[Z_AXIS]);
+//	    THEKERNEL->streams->printf(" CalcZ= %f\n",z_calc_offset);
+//	    THEKERNEL->streams->printf(" Offz= %f\n",z_with_offset);
+	  }
+	  else
+	  {
+	    actuator_mm[ALPHA_STEPPER] = alpha_theta;
+	    actuator_mm[BETA_STEPPER ] = beta_theta;
+	    actuator_mm[GAMMA_STEPPER] = gamma_theta;
+
+//		  THEKERNEL->streams->printf("cartesian x= %f\n\r",cartesian_mm[X_AXIS]);
+//		  THEKERNEL->streams->printf(" y= %f\n\r",cartesian_mm[Y_AXIS]);
+//		  THEKERNEL->streams->printf(" z= %f\n\r",cartesian_mm[Z_AXIS]);
+//		  THEKERNEL->streams->printf(" Offz= %f\n\r",z_with_offset);
+//		  THEKERNEL->streams->printf(" delta x= %f\n\r",delta[X_AXIS]);
+//		  THEKERNEL->streams->printf(" y= %f\n\r",delta[Y_AXIS]);
+//		  THEKERNEL->streams->printf(" z= %f\n\r",delta[Z_AXIS]);
+	  }
+
+}
+
+void RotatableDeltaSolution::actuator_to_cartesian( float actuator_mm[], float cartesian_mm[] )
+{
+    //Use forward kinematics
+    delta_calcForward(actuator_mm[ALPHA_STEPPER], actuator_mm[BETA_STEPPER ], actuator_mm[GAMMA_STEPPER], cartesian_mm[X_AXIS], cartesian_mm[Y_AXIS], cartesian_mm[Z_AXIS]);
+}
+
+bool RotatableDeltaSolution::set_optional(const arm_options_t& options) {
+
+    for(auto &i : options) {
+        switch(i.first) {
+            case 'A': delta_e			= i.second; break;
+            case 'B': delta_f			= i.second; break;
+            case 'C': delta_re			= i.second; break;
+            case 'D': delta_rf			= i.second; break;
+            case 'E': delta_z_offset		= i.second; break;
+            case 'F': delta_ee_offs		= i.second; break;
+            case 'G': tool_offset		= i.second; break;
```

Comment:
Ah, I see why, as these are reserved for the G T M & N codes :-)

I have fixed that.
> On 11/06/2015, at 10:46 am, Jim Morris <notifications@github.com> wrote:
> 
> In src/modules/robot/arm_solutions/RotatableDeltaSolution.cpp <https://github.com/Smoothieware/Smoothieware/pull/673#discussion_r32174782>:
> 
> > +{
> > +    //Use forward kinematics
> > +    delta_calcForward(actuator_mm[ALPHA_STEPPER], actuator_mm[BETA_STEPPER ], actuator_mm[GAMMA_STEPPER], cartesian_mm[X_AXIS], cartesian_mm[Y_AXIS], cartesian_mm[Z_AXIS]);
> > +}
> > +
> > +bool RotatableDeltaSolution::set_optional(const arm_options_t& options) {
> > +
> > +    for(auto &i : options) {
> > +        switch(i.first) {
> > +            case 'A': delta_e			= i.second; break;
> > +            case 'B': delta_f			= i.second; break;
> > +            case 'C': delta_re			= i.second; break;
> > +            case 'D': delta_rf			= i.second; break;
> > +            case 'E': delta_z_offset		= i.second; break;
> > +            case 'F': delta_ee_offs		= i.second; break;
> > +            case 'G': tool_offset		= i.second; break;
> G is not a valid parameter (neither is T or M or N)
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/Smoothieware/Smoothieware/pull/673/files#r32174782>.
>